server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java

```diff
@@ -81,10 +124,26 @@ public class DiskThresholdSettings {
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
+    private static final List<Setting<?>> WATERMARK_VALIDATOR_SETTINGS_LIST = List.of(
+        CLUSTER_ROUTING_ALLOCATION_LOW_DISK_WATERMARK_SETTING,
+        CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_WATERMARK_SETTING,
+        CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_WATERMARK_SETTING
+    );
+    private static final List<Setting<?>> MAX_HEADROOM_VALIDATOR_SETTINGS_LIST = List.of(
+        CLUSTER_ROUTING_ALLOCATION_LOW_DISK_MAX_HEADROOM_SETTING,
+        CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_MAX_HEADROOM_SETTING,
+        CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_MAX_HEADROOM_SETTING,
+        CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_FROZEN_MAX_HEADROOM_SETTING,
+        CLUSTER_ROUTING_ALLOCATION_LOW_DISK_WATERMARK_SETTING,
+        CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_FROZEN_WATERMARK_SETTING
```

Comment:
Are we not missing `CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_WATERMARK_SETTING`? If so, some test must be missing too?

Reply:
Hi @henningandersen , these are the settings on which the org.elasticsearch.cluster.routing.allocation.DiskThresholdSettings.MaxHeadroomValidator#validate() depends on. The validator is checking the headroom values primarily, but also checks that if the headroom values are set, then watermark values are ratios/percentages. To check whether the watermark values are ratios/percentages, we only need to check one of them (the low watermark in this case), since the other `WatermarkValidator` checks that if at least one of the watermark values are ratios/percentages, all of them (low, high, flood) need to be ratios/percentages. That is why there is no need for the max headroom validator to depend on all 3 watermark settings, just one of them. And there is a similar check for the independent frozen headroom (that if it is set, then the frozen watermark is a ratio/percentage).

There are also relevant tests org.elasticsearch.cluster.routing.allocation.DiskThresholdSettingsTests#testIncompatibleMaxHeadroomUpdate and testIncompatibleFrozenMaxHeadroomUpdate for these.

Reply:
I will resolve this as I think I gave enough explanation. Please unresolve the conversation if it is not sufficient.

Reply:
Can we add a short comment to the code around why they are missing (not the entire blurb here, just short explanation to help the code reader)?

Reply:
Sure, you will see in next commit. When you check it and feel it's not sufficient, please unresolve this conversation.